openshift/machine-config-operator
# MCO-1227: Run boot image controller only when required

**- What I did**
This PR ensures that the boot image controller only runs on the correct combination of platform and feature gates.

**- How to verify it**
Bring up a cluster in techpreview. The boot image controller should only run on GCP and AWS platforms. This can be verified from the machine-config-controller logs.
``` 
oc logs -n openshift-machine-config-operator -f "$(oc get pod -o name -l='k8s-app=machine-config-controller' -n openshift-machine-config-operator)" | grep machine_set
```
Output should look something like:
```
I0717 14:13:22.580249       1 machine_set_boot_image_controller.go:275] Bootimages management configuration has been added, reconciling enrolled machine resources
I0717 14:13:22.580273       1 machine_set_boot_image_controller.go:361] No MAPI machinesets were enrolled, so no MAPI machinesets will be enqueued.
I0717 14:13:22.581460       1 machine_set_boot_image_controller.go:171] MAPI MachineSet djoshy20-rwm98-worker-a added, reconciling enrolled machine resources
I0717 14:13:22.581541       1 machine_set_boot_image_controller.go:171] MAPI MachineSet djoshy20-rwm98-worker-b added, reconciling enrolled machine resources
I0717 14:13:22.581576       1 machine_set_boot_image_controller.go:171] MAPI MachineSet djoshy20-rwm98-worker-c added, reconciling enrolled machine resources
I0717 14:13:22.582006       1 machine_set_boot_image_controller.go:221] configMap coreos-bootimages added, reconciling enrolled machine resources
I0717 14:13:22.661888       1 machine_set_boot_image_controller.go:161] Starting MachineConfigController-MachineSetBootImageController
I0717 14:13:22.730355       1 machine_set_boot_image_controller.go:361] No MAPI machinesets were enrolled, so no MAPI machinesets will be enqueued.
I0717 14:13:23.021122       1 machine_set_boot_image_controller.go:361] No MAPI machinesets were enrolled, so no MAPI machinesets will be enqueued.
I0717 14:13:23.404167       1 machine_set_boot_image_controller.go:361] No MAPI machinesets were enrolled, so no MAPI machinesets will be enqueued.
I0717 14:13:23.789165       1 machine_set_boot_image_controller.go:361] No MAPI machinesets were enrolled, so no MAPI machinesets will be enqueued.
```
On the default feature set, or any other platform in tech preview, the boot image controller should not come up, so no logs should show up.

Note: This PR only affects the enabling/disabling of the boot image controller - it does not affect its functionality. It should pass the existing e2es.
